main/src/observables/gravitational_waves.hpp

```diff
@@ -0,0 +1,195 @@
+/*
+* MIT License
+*
+* Copyright (c) 2021 CSCS, ETH Zurich
+*               2021 University of Basel
+*
+* Permission is hereby granted, free of charge, to any person obtaining a copy
+* of this software and associated documentation files (the "Software"), to deal
+* in the Software without restriction, including without limitation the rights
+* to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+* copies of the Software, and to permit persons to whom the Software is
+* furnished to do so, subject to the following conditions:
+*
+* The above copyright notice and this permission notice shall be included in all
+* copies or substantial portions of the Software.
+*
+* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+* IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+* FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+* AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+* LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+* OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+* SOFTWARE.
+*/
+
+/*! @file calculate gravitational waves observable
+ *
+ *  @author Lukas Schmidt
+ */
+
+#include <array>
+#include "mpi.h"
+
+#include "iobservables.hpp"
+#include "io/ifile_writer.hpp"
+
+namespace sphexa
+{
+
+//!@brief calculates the second derivative of the quadpole momentum
+template<class T, class Dataset>
+T d2QuadpoleMomentum(size_t begin, size_t end, int pos1, int pos2, Dataset& d)
```

Comment:
This should only take the arrays that it needs listed below, rather than the complete Dataset.
grad_P should actually be called acceleration (ax, ay, az) here and you will have to change the sign once https://github.com/unibas-dmi-hpc/SPH-EXA/pull/252 is merged

Reply:
dim1 and dim2 would be better names for pos1 and pos2

Reply:
the argument arrays, x, v, accel should be passed as `const`